File: android/app/src/main/java/com/banshee/translatekeyboard/translate/Translator.kt
package com.banshee.translatekeyboard.translate

import com.banshee.translatekeyboard.settings.TargetLang

class Translator(
    private val deepL: DeepLClient,
    private val myMemory: MyMemoryClient = MyMemoryClient(),
) {
    suspend fun translate(text: String, target: TargetLang): Result<String> {
        val deeplCode = target.deeplCode
        return if (deeplCode != null) {
            deepL.translate(text, deeplCode)
        } else {
            val source = SundaneseHeuristics.guessSourceFor(text)
            myMemory.translate(text, source, target.myMemoryCode)
        }
    }
}

File: ios/Shared/Translator.swift
import Foundation

struct Translator {
    var deepL = DeepLClient()
    var myMemory = MyMemoryClient()

    func translate(_ text: String, target: TargetLang) async throws -> String {
        if let deeplCode = target.deeplCode {
            _ = deeplCode
            return try await deepL.translate(text, targetLang: target)
        } else {
            let source = SundaneseHeuristics.guessSourceFor(text)
            return try await myMemory.translate(text, sourceLang: source, targetLang: target.myMemoryCode)
        }
    }
}
